File: Buffet_a_volonte/process.rb
#*******
#* Read input from STDIN
#* Use puts to output your result
#* Use:
#*      local_print( variable ); 
#* to display simple variables in a dedicated area.
#* 
#* ***/
input = Array.new
while i = $stdin.gets
	input << i.strip
end

prix_repas = Integer(input[0])
tables = input[2,input.size].map { |number| Integer(number) }

accumulator = 0
tables.each do |table|
	pourcentage = 0
	if table >= 10
		pourcentage = 0.3
	elsif table >= 6
		pourcentage = 0.2
	elsif table >= 4
		pourcentage = 0.1
	end

	prix_fix = prix_repas * table
	accumulator += prix_fix - (prix_fix * pourcentage)
end

puts(accumulator.ceil)

#
# perform your processing and output your result...
#

File: Buffet_a_volonte/process.py
#*******
#* Read input from STDIN
#* Use echo or print to output your result, use the /n constant at the end of each result line.
#* Use:
#*      local_print (variable ); 
#* to display simple variables in a dedicated area.
#* ***/
import sys
import math

lines = []
for line in sys.stdin:
	lines.append(line.rstrip('\n'))

prix_repas = int(lines[0])
prix_tables = list()
tables = [int(val) for val in lines[2:]]
for table in tables:
    reduction = 0
    if table >= 10:
        reduction = 0.3
    elif table >= 6:
        reduction = 0.2
    elif table >= 4:
        reduction = 0.1
    prix_table = table * prix_repas 
    prix_tables.append(prix_table - (reduction * prix_table))
resultat = math.ceil(sum(map(float,prix_tables)))
print(resultat)
